tests/lsp_features/hover.zig

```diff
@@ -464,15 +499,80 @@ test "hover - integer overflow on top level container" {
     );
 }
 
+test "hover - combine doc comments of declaration and definition" {
+    try testHover(
+        \\/// Foo
+        \\const f<cursor>oo = bar.baz;
+        \\const bar = struct {
+        \\    /// Bar
+        \\    const baz = struct {};
+        \\};
+    ,
+        \\```zig
+        \\const baz = struct
+        \\```
+        \\```zig
+        \\(type)
+        \\```
+        \\
+        \\ Foo
+        \\
+        \\ Bar
+    );
+    try testHoverWithOptions(.{ .markup_kind = .plaintext },
+        \\/// Foo
+        \\const f<cursor>oo = bar.baz;
+        \\const bar = struct {
+        \\    /// Bar
+        \\    const baz = struct {};
+        \\};
+    ,
+        \\const baz = struct
+        \\(type)
+        \\
+        \\ Foo
+        \\
+        \\ Bar
+    );
+}
+
+test "hover - top-level doc comment" {
+    try testHover(
+        \\//! B
+        \\
+        \\/// A
+        \\const S<cursor>elf = @This();
+    ,
+        \\```zig
+        \\const Self = @This()
+        \\```
+        \\```zig
+        \\(type)
+        \\```
+        \\
+        \\ A
+        \\
+        \\ B
+    );
+}
+
 fn testHover(source: []const u8, expected: []const u8) !void {
+    try testHoverWithOptions(.{ .markup_kind = .markdown }, source, expected);
+}
+
+fn testHoverWithOptions(
+    options: struct { markup_kind: types.MarkupKind },
```

Comment:
could you move this parameter to the end to be more consistent with the other feature test functions.